File: apps/cinder/src/handlers/metric.ts
import type { CustomMetric } from '@bx-team/types/schema/pulsify';
import type { Env } from '../env';
import { writeMetric } from '../lib/analytics';

export function handleMetric(env: Env, event: CustomMetric, projectId: string) {
  writeMetric(env, projectId, {
    name: event.name,
    value: event.value,
    labels: event.labels ?? {},
  });
}

File: apps/cinder/src/handlers/metric.rs
use analytics::writer::MetricPoint;
use database::queue::QueuedEvent;
use types::ingest::MetricEvent;

use super::Outcome;

pub fn handle(message: &QueuedEvent, event: &MetricEvent, outcome: &mut Outcome) {
    outcome.analytics.metrics.push(MetricPoint {
        timestamp: message.received_at,
        project_id: message.project_id,
        name: event.name.clone(),
        labels: event.labels.clone(),
        value: event.value,
    });
}
